crates/host/src/wasmbus/mod.rs

```diff
@@ -2374,108 +2215,93 @@ impl Host {
             origin,
             cluster_key: Arc::clone(&self.cluster_key),
             claims: claims.clone(),
-            aliases: Arc::clone(&self.aliases),
-            links: Arc::new(RwLock::new(links)),
             interface_link_name: Arc::new(RwLock::new("default".to_string())),
             interface_links: Arc::new(RwLock::new(component_spec.links)),
             host_key: Arc::clone(&self.host_key),
             polyfilled_imports: imports,
         };
 
-        let instance = self
+        let actor = self
             .instantiate_actor(
-                claims,
                 &annotations,
                 &actor_ref,
+                &actor_id,
                 max_instances,
-                actor.clone(),
+                component.clone(),
                 handler.clone(),
             )
             .await
             .context("failed to instantiate actor")?;
 
         info!(actor_ref, "actor started");
-        self.publish_actor_started_events(
-            max_instances.get(),
-            max_instances,
-            claims,
-            &annotations,
-            host_id,
-            actor_ref,
+        self.publish_event(
+            "actor_scaled",
+            event::actor_scaled(claims, &annotations, host_id, max_instances, &actor_ref),
         )
         .await?;
 
-        let actor = Arc::new(Actor {
-            actor,
-            instances: RwLock::new(HashMap::from([(annotations, instance)])),
-            handler,
-        });
         Ok(entry.insert(actor))
     }
 
     #[instrument(level = "debug", skip_all)]
-    async fn stop_actor<'a>(
-        &self,
-        entry: hash_map::OccupiedEntry<'a, String, Arc<Actor>>,
-        annotations: &BTreeMap<String, String>,
-        host_id: &str,
-    ) -> anyhow::Result<()> {
-        trace!(actor_id = %entry.key(), "stopping actor");
-
-        let actor = entry.remove();
-        let claims = actor.claims().context("claims missing")?;
-        let mut instances = actor.instances.write().await;
+    async fn stop_actor(&self, actor: Arc<Actor>, host_id: &str) -> anyhow::Result<()> {
```

Comment:
Note: I think this can take a borrowed actor (as long as `actor.calls.abort()` is `&self`, which I think it is). Feels a little odd to clone the arc in order to stop the actor

Reply:
Maybe we can just move it 🤔 if we're stopping it we shouldn't use it afterwards